### DISCUSSION TITLE
Redhat 8 Installation using EFI based installation, After completed the boot command with Ctrl-x, VM goes to poweroff state

### DISCUSSION THREAD

**Discussion by prabahar-subramani** [hashicorp/packer-plugin-vsphere#648] [Community Help]
Hi All,

I'm trying to provision EFI based Redhat OS installation using packer with Vcenter. Here i'm using Kickstart file for preseeding. After typed the Boot Command and then do the Ctrl-X and then VM goes to poweroff state.

Below the Packer Information:
boot_order = "disk,cdrom,floppy"
  boot_wait  = "10s"
  boot_command = [
    "<up><wait>",
    "e<wait>",
    "<down><down>",
    "<del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del><del>",
    "<spacebar><spacebar><spacebar><spacebar><spacebar><spacebar><spacebar><spacebar>linuxefi /images/pxeboot/vmlinuz inst.ks=hd:sr1:ks.cfg inst.stage2=hd:LABEL=RHEL-8-6-0-BaseOS-x86_64 quiet",
    "<wait10>",
    "<leftCtrlOn>x<leftCtrlOff>",
    "<wait900>"
  ]

Packer Version:
Packer v1.8.0

And also tried on Packer : 1.9.0 

Still the issue exist.

**Comment by tenthirtyam** [comment]
@lbajolet-hashicorp - could you transfer this issue to `hashicorp/packer-plugin-vsphere`, please? Thanks!

**Comment by tenthirtyam** [comment]
Hi @prabahar-subramani 

The following `boot_command` can be used recommended on RHEL 8 and 9 (and related variants:

**HTTP-based**:

```hcl
  boot_command = [
    // This sends the "up arrow" key, typically used to navigate through boot menu options.
    "<up>",
    // This sends the "e" key. In the GRUB boot loader, this is used to edit the selected boot menu option.
    "e",
    // This sends two "down arrow" keys, followed by the "end" key, and then waits. This is used to navigate to a specific line in the boot menu option's configuration.
    "<down><down><end><wait>",
    // This types the string "text" followed by the value for the method used.
    // This is used to modify the boot menu option's configuration to boot in text mode and specify the kickstart data source configured in the common variables.
    "text inst.ks=http://{{ .HTTPIP }}:{{ .HTTPPort }}/ks.cfg",
    // This sends the "enter" key, waits, turns on the left control key, sends the "x" key, and then turns off the left control key. This is used to save the changes and exit the boot menu option's configuration, and then continue the boot process.
    "<enter><wait><leftCtrlOn>x<leftCtrlOff>"
  ]
```

**CD-based**:

```hcl
  boot_command = [
    // This sends the "up arrow" key, typically used to navigate through boot menu options.
    "<up>",
    // This sends the "e" key. In the GRUB boot loader, this is used to edit the selected boot menu option.
    "e",
    // This sends two "down arrow" keys, followed by the "end" key, and then waits. This is used to navigate to a specific line in the boot menu option's configuration.
    "<down><down><end><wait>",
    // This types the string "text" followed by the value for the method used.
    // This is used to modify the boot menu option's configuration to boot in text mode and specify the kickstart data source configured in the common variables.
    "text inst.ks=cdrom:/ks.cfg",
    // This sends the "enter" key, waits, turns on the left control key, sends the "x" key, and then turns off the left control key. This is used to save the changes and exit the boot menu option's configuration, and then continue the boot process.
    "<enter><wait><leftCtrlOn>x<leftCtrlOff>"
  ]
```

**Comment by tenthirtyam** [comment]
Marking as closed. Solution provided in https://github.com/hashicorp/packer-plugin-vsphere/issues/426#issuecomment-2099120847.
